src/MatrixProductOperator.jl

```diff
@@ -0,0 +1,175 @@
+using OptimizedEinsum: ContractionPath
+using UUIDs: uuid4
+using IterTools: partition
+using Random
+
+abstract type MatrixProductOperator{B} <: State{B} end
+
+function MatrixProductOperator(arrays; bounds::Type{<:Bounds} = Open, kwargs...)
+    MatrixProductOperator{bounds}(arrays; kwargs...)
+end
+
+function MatrixProductOperator{Open}(arrays; χ = nothing, order = (:l, :r, :p), meta...)
+    !issetequal(order, (:l, :r, :p)) && throw(ArgumentError("`order` must be a permutation of the :l, :r and :p"))
+    order = Dict(dim => i for (i, dim) in enumerate(order))
```

Comment:
Where do you check for input/output index order? Here you only have 1 physical index.

Reply:
Ups. Missed that. Thanks!